server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateShardPathIT.java

```diff
@@ -21,10 +22,17 @@
 
 import java.util.HashSet;
 import java.util.Set;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
 import static org.hamcrest.Matchers.equalTo;
 
+/*
+ * We rely on the shard directory being deleted after the relocation. This removal sometimes fails
+ * with "java.io.IOException: access denied" when using WindowsFS which seems to be a known issue.
+ * See {@link FileSystemUtilsTests}.
+ */
+@LuceneTestCase.SuppressFileSystems(value = "WindowsFS")
```

Comment:
I wonder if the retrying below would not cure this regardless? IIUC, we catch such failures to delete and retry later.

If the API is not supported on windows, we should probably note it in the documentation.

Reply:
Just retrying is still not enough and sometimes the test would still failed with "access denied exception". Which is why I ended up adding the suppress WindowsFS bit.

> If the API is not supported on windows, we should probably note it in the documentation.

If you mean the prevalidation API, I am not sure this has any implications for that. We only added this part (ensuring that the shards are removed after relocation) to just make sure [our assumptions are correct](https://github.com/elastic/elasticsearch/pull/91256#discussion_r1031386237). Which I think is, but we need this fix to make the test pass.

Reply:
OK, though if we sometimes leave stuff around on windows, in essence this API may not work as reliably there. But I guess that is more of a problem of the relocation than this API.

Reply:
I think the API does what it should do. As long as it sees the shard directory it reports back so. Yes, I agree that the lingering existence of the directory is more a relocation issue, but to me it seems that also is retried in a reconciliation fashion upon cluster state changes. What would make the result of the prevalidation less likely to change quickly after responding is [this improvement that we talked about](https://github.com/elastic/elasticsearch/pull/91256#issuecomment-1306976226), which I'm still to write up as an issue.